Introduce cross-cluster replication API docs (#34726)
elastic/elasticsearch: docs/reference/ccr/apis/follow/put-follow.asciidoc

[role="xpack"]
[testenv="platinum"]
[[ccr-put-follow]]
=== Create Follower API
++++
<titleabbrev>Create Follower</titleabbrev>
++++

Creates a follower index.

==== Description

This API creates a new follower index that is configured to follow the
referenced leader index. When this API returns, the follower index exists, and

> {ccr} will start replicating operations from the leader index to the follower

Reviewer: ```suggestion
{ccr} starts replicating operations from the leader index to the follower
```

index.

==== Request

//////////////////////////

[source,js]
--------------------------------------------------
POST /follower_index/_ccr/pause_follow
--------------------------------------------------
// CONSOLE
// TEARDOWN

//////////////////////////

[source,js]
--------------------------------------------------
PUT /<follower_index>/_ccr/follow
{
  "leader_cluster" : "<leader_cluster>",
  "leader_index" : "<leader_index>"
}
--------------------------------------------------
// CONSOLE
// TEST[setup:leader_cluster_and_leader_index]
// TEST[s/<follower_index>/follower_index/]
// TEST[s/<leader_cluster>/leader_cluster/]
// TEST[s/<leader_index>/leader_index/]

Here, `<follower_index>` is the name of the follower index to create,
`<leader_cluster>` is the name of the
<<modules-remote-clusters,remote cluster> containing the leader index, and
`<leader_index>` is the name of the leader index in that remote cluster.

> 

Reviewer: ```suggestion

==== Path Parameters

`follower_index` (required)::
  (string) Identifier for the follower index.

```

==== Request Body

> `leader_cluster`::

Reviewer: ```suggestion
`leader_cluster` (required)::
```

>   (required string) the <<modules-remote-clusters,remote cluster>> containing the leader

Reviewer: ```suggestion
  (string) The <<modules-remote-clusters,remote cluster>> containing the leader
```

  index

> `leader_index`::

Reviewer: ```suggestion
`leader_index` (required)::
```

>   (required string) the name of the index in the leader cluster to follow

Reviewer: ```suggestion
  (string) The name of the index in the leader cluster to follow
```

include::../follow-request-body.asciidoc[]

==== Example

> The following is an example of using the create follower API.

Reviewer: ```suggestion
This example creates a follower index named `follower_index`:
```
